packages/angular/projects/clr-angular/src/data/datagrid/datagrid-column-separator.ts

```diff
@@ -4,74 +4,160 @@
  * The full license information can be found in LICENSE in the root directory of this project.
  */
 import { DOCUMENT } from '@angular/common';
-import { Component, Inject, Renderer2 } from '@angular/core';
-
-import { ClrDragEvent } from '../../utils/drag-and-drop/drag-event';
+import { AfterViewInit, Component, ElementRef, Inject, NgZone, OnDestroy, Renderer2, ViewChild } from '@angular/core';
+import { KeyCodes, IEKeyCodes } from '../../utils/enums/key-codes.enum';
+import { ClrCommonStringsService } from '../../utils/i18n/common-strings.service';
+import { UNIQUE_ID, UNIQUE_ID_PROVIDER } from '../../utils/id-generator/id-generator.service';
 import { ColumnResizerService } from './providers/column-resizer.service';
 import { TableSizeService } from './providers/table-size.service';
-import { UNIQUE_ID_PROVIDER, UNIQUE_ID } from '../../utils/id-generator/id-generator.service';
+
+// Default resize length on each keyboard move event
+const KEYBOARD_RESIZE_LENGTH = 12;
 
 @Component({
   selector: 'clr-dg-column-separator',
   template: `
-    <div
+    <button
+      type="button"
       class="datagrid-column-handle"
-      aria-hidden="true"
+      [attr.aria-label]="commonString.keys.columnSeparatorAriaLabel"
+      [attr.aria-describedby]="descriptionId"
       clrDraggable
       [clrGroup]="columnSeparatorId"
-      (clrDragStart)="showTracker(resizeTrackerEl)"
-      (clrDragMove)="moveTracker($event, resizeTrackerEl)"
-      (clrDragEnd)="hideTracker(resizeTrackerEl)"
-    ></div>
-    <div class="datagrid-column-resize-tracker" #resizeTrackerEl></div>
+      (clrDragStart)="showTracker()"
+      (clrDragMove)="moveTracker($event.dragPosition.moveX)"
+      (clrDragEnd)="hideTracker()"
+      #columnHandle
+    ></button>
+    <span class="clr-sr-only" [attr.id]="descriptionId">
+      {{ commonString.keys.columnSeparatorDescription }}
+    </span>
+    <div class="datagrid-column-resize-tracker" #resizeTracker></div>
   `,
   host: {
     '[class.datagrid-column-separator]': 'true',
   },
   providers: [UNIQUE_ID_PROVIDER],
 })
-export class ClrDatagridColumnSeparator {
+export class ClrDatagridColumnSeparator implements AfterViewInit, OnDestroy {
+  @ViewChild('resizeTracker', { static: false })
+  resizeTrackerRef: ElementRef;
+
+  get resizeTrackerEl() {
+    return this.resizeTrackerRef.nativeElement;
+  }
+
+  @ViewChild('columnHandle', { static: false })
+  columnHandleRef: ElementRef;
+
+  get columnHandleEl() {
+    return this.columnHandleRef.nativeElement;
+  }
+
+  private resizeStartedOnKeyDown = false;
+
+  private isWithinMaxResizeRange: boolean;
+
+  private unlisteners: (() => void)[] = [];
```

Comment:
Why is this `unlisteners` instead of `subscriptions` or `listeners`?

Reply:
This array holds callback methods that remove their listeners.  So it describes what it does better than `subscriptions` or `listeners`, I think.